ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -18,30 +19,66 @@
-        _ (acl/verify-ingest-management-permission
-            request-context :update :provider-object provider-id)
-        _ (acl/verify-subscription-management-permission
-            request-context :update :provider-object provider-id)
-        _ (common-enabled/validate-write-enabled request-context "ingest")
-        concept (validate-and-prepare-subscription-concept concept)
-        concept-with-user-id (api-core/set-user-id concept request-context headers)
-        ;; Log the ingest attempt
+(defn- perform-subscription-ingest
+  "This function assumes all checks have already taken place and that a
+  subscription is ready to be saved"
+  [concept request-context headers]
+  (let [validated-concept (validate-and-prepare-subscription-concept concept)
+        concept-with-user-id (api-core/set-user-id validated-concept
+                                                   request-context
+                                                   headers)
+        ;; Log the ingest attempt before the save
         _ (info (format "Ingesting subscription %s from client %s"
                         (api-core/concept->loggable-string concept-with-user-id)
                         (:client-id request-context)))
-        save-subscription-result (ingest/save-subscription request-context concept-with-user-id)]
+        save-subscription-result (ingest/save-subscription request-context
+                                                           concept-with-user-id)]
     ;; Log the successful ingest, with the metadata size in bytes.
     (api-core/log-concept-with-metadata-size concept-with-user-id request-context)
     (api-core/generate-ingest-response headers save-subscription-result)))
 
+(defn- check-subscription-ingest-permission
+  "All the checks needed before starting to process an ingest of subscriptions"
+  [request-context provider-id]
+  (common-enabled/validate-write-enabled request-context "ingest")
+  (lt-validation/validate-launchpad-token request-context)
+  (api-core/verify-provider-exists request-context provider-id))
+
+(defn ingest-subscription
+  "Processes a request to create or update a subscription. Note, this will allow
+  unlimited subscriptions to be created by users for themselves. Revisit if this
+  Becomes a problem"
+  [provider-id native-id request]
+  (let [{:keys [body content-type headers request-context]} request]
+    (check-subscription-ingest-permission request-context provider-id)
+    (let [concept (api-core/body->concept! :subscription provider-id native-id
+                                           body
+                                           content-type
+                                           headers)
+          subscription-user (:SubscriberId (json/decode (:metadata concept) true))
+          token-user (api-core/get-user-id request-context headers)]
+      (if (and token-user ; don't allow a bypass of ACLs just because of nil values
```

Comment:
My review comment was to move these user-id and ACL checks into a function called `check-subscription-ingest-permission`, not the three basic checks that you have moved. Those three basic checks are OK to be listed as is in this `ingest-subscription` function.